### ISSUE TITLE
oom killed at the second epoch to save checkpoint(stable duffusion v2)

### ISSUE THREAD

**Issue by yufengyao-lingoace** [hpcaitech/ColossalAI#2398]
I'm training stable diffusion v2. The first epoch is normal and can export a ckpt file, but the second epoch can not save checkpoints. It used a large amount of cpu cache, and then be killed. My machine has 32G cpu memory and 16g gpu memory, and 20g virtual cpu memory. 

Epoch 0: 100%|█| 163/163 [18:08<00:00,  6.68s/it, loss=0.992, v_num=0, train/loss_simple_step=0.990, train/loss_vlb_step=0.00506, train/loss_step=0.990, global_step=162.0, lr_abs=1.6e-7, t/home/ubuntu/anaconda3/envs/ldmco/lib/python3.9/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:359: UserWarning: `ModelCheckpoint(monitor='val/loss_simple_ema')` could not find the monitored key in the returned metrics: ['train/loss_simple', 'train/loss_simple_step', 'train/loss_vlb', 'train/loss_vlb_step', 'train/loss', 'train/loss_step', 'global_step', 'lr_abs', 'train/loss_simple_epoch', 'train/loss_vlb_epoch', 'train/loss_epoch', 'epoch', 'step']. HINT: Did you call `log('val/loss_simple_ema', value)` in the `LightningModule`?
  warning_cache.warn(m)
Epoch 1: 100%|█| 163/163 [18:20<00:00,  6.75s/it, loss=0.98, v_num=0, train/loss_simple_step=0.976, train/loss_vlb_step=0.00521, train/loss_step=0.975, global_step=325.0, lr_abs=1.6e-7, tr
Killed

**Comment by yufengyao-lingoace** [comment]
It seems to be a memory leak.
